src/WebJobs.Script.WebHost/Middleware/FunctionInvocationMiddleware.cs

```diff
@@ -67,6 +67,11 @@ public async Task Invoke(HttpContext context)
                     HttpContext = context
                 };
 
+                if (result is RedirectResult)
+                {
+                    actionContext.RouteData = new RouteData();
```

Comment:
You only want to do this if RouteData is null, otherwise we may be overwriting unintentionally?

Reply:
Also, add a comment here that we're working around a regression, linking to the .NET Core issue

Reply:
Also, recommend adding a test for this, e.g. to the scenarios integration tests https://github.com/Azure/azure-functions-host/blob/dev/test/WebJobs.Script.Tests.Integration/TestScripts/CSharp/Scenarios/run.csx

Reply:
I'd recommend doing this all the time. The constructor under use isn't meant for not-test scenarios (https://docs.microsoft.com/en-us/dotnet/api/microsoft.aspnetcore.mvc.actioncontext.-ctor?view=aspnetcore-2.1#Microsoft_AspNetCore_Mvc_ActionContext__ctor). 

```C#
var actionContext = new ActionContext(context, new RouteData(), new ActionDescriptor());
```

Generally, running an `ActionResult` outside of the context of MVC action isn't a supported scenario. I'd recommend testing against our pre-release binaries to anticipate issues like these early on.